### DISCUSSION TITLE
Startup Assist Level & Temperature in Walking Mode

### DISCUSSION THREAD

**Discussion by Bellatrix92** [danielnilsson9/bbs-fw#249] [General]
Hi!
I will just start by applauding Daniel for his amazing job!

1. I am wondering if someone with BBS02 has been able to succesfully set the startup assist level and make it work? 
It is always starting in level 1 for me even though I set it to anything else in the BBS-FW tool.

2. I am also unable to see temperature in walking mode..

I have 3 controllers, different displays and they all behave the same.

Grateful for any inputs here.

Thanks :)

**Comment by WvBijsterveld** [comment]
i think your display overrides the startup assist.

Regarding your temperature, ensure your have set the motor temperature off because if i recall correctly only the controller temp sensor is in the bbs02. 

or go to the log tab in the config tool and see if your temp sensors are working at all.

**Reply by Bellatrix92** [reply]
Thanks for replying,

Yes it might be the displays overriding somehow. But the same display with bbshd and original firmware always start in assist 0. So it is a bit strange.

Yes temperature is only in controller I think. It works in the log. I set it to all in FW tool. Maybe I should change it to Controller.

**EDIT**: Temperature is now working :).
Then only remaining "issue" for me is startup assist. I will fiddle around with different displays and see if it makes any difference.

**Comment by aalm** [comment]
Display and the firmware have nothing in the protocol to allow the controller to let display know which level it is in.
So display will always show what it does believe it is(possibly configurable from display settings), and i believe some
displays just "spam" the wanted state to the controller even if nothing has changed.

I think the startup assist level setting is for use w/o display due what i wrote above.

**Reply by aalm** [reply]
I want to add, that i don't have experience of many displays(only two different kinds, and the first one is already broken),
but i have recently written my own copy to a nearly complete state, so i'm rather familiar with displays by now anyway;
![2024-11-14_706x605](https://github.com/user-attachments/assets/b1d25bb3-5cd3-4bef-9b3e-860cf43b0e66)

 ^is already usable for changing assist level, mode etc., and i will add the missing walkmode-switch next:]

**Comment by danielnilsson9** [comment]
As others have said, display will override controller startup assist level. The setting is intended for when running without a display or as fallback if communication with display is broken (error 30),

Documented here:
https://github.com/danielnilsson9/bbs-fw/wiki/Configuration-Tool#assist-levels
